core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanOps.scala

```diff
@@ -105,9 +136,3 @@ trait SpanOps[F[_]] {
     */
   def surround[A](fa: F[A]): F[A]
```

Comment:
this can still be `final ... = use(_ => fa)`, will change momentarily

Reply:
It's not final on `Resource` itself.  I wonder whether that implies possible optimizations.  But I can't imagine any offhand.

Reply:
technically you can optimise the no-op implementation slightly by not creating a wrapper `Function1` around the `F[A]` to pass to `use`, but honestly I still feel it's cleaner to have it be final. it can be made not final in a minor release if minds are changed later. or someone can make a PR after this is merged. or if others also want it non-final, I can add another fixup commit to this PR